src/support/hazard.c

```diff
@@ -350,6 +350,72 @@ __wt_hazard_check(WT_SESSION_IMPL *session, WT_REF *ref, WT_SESSION_IMPL **sessi
     return (hp);
 }
 
+/*
+ * __wt_hazard_check_more_than_one --
+ *     Return if there are more than one hazard pointers to the page in the system.
+ */
+bool
+__wt_hazard_check_more_than_one(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_HAZARD *hp;
+    WT_SESSION_IMPL *s;
+    size_t count;
+    uint32_t i, j, hazard_inuse, max, session_cnt, walk_cnt;
+
+    /* If a file can never be evicted, hazard pointers aren't required. */
+    if (F_ISSET(S2BT(session), WT_BTREE_IN_MEMORY))
+        return (false);
```

Comment:
I made a mistake here. I think we should return true here to stop it from evicting. This prompts me to think we probably should rename this function to something __wt_hazard_release_can_evict?